Component Replacement

RayRedGoose opened the pull request:
### This PR contains the next changes:

- Replaced App, DisplayInsights, FinalIdeaField, Header, SecretSauce, Timer, UserProfile components to Containers folder because they have a connection with Redux store.
- Replaced WelcomePage, IdeaEventHorizon, SignUpForm components to Components folder because they are not connected with Redux store.
- Added redirection to Welcome page from other URLs if there is no user logged.
- Removed unused variables.
- Added missed type checks.

JamesRexMiller4 commented:
Hey I noticed that the insights div I was working on in RoundTwo still has some buggy styles. Particularly if you hold down one letter on the keyboard with no spaces, it will break out of the containing div width. Can you take a look at that when you get a chance?

JamesRexMiller4 commented:
Also what do you think about allowing the typed input to remain as it goes from round 1 to 2, 2 to 3 for RoundTwo. Part of me was thinking of clearing the input automatically, but then another part of me was thinking that if they had typed up a question they really liked and then it was removed that might be worse than them having to delete it themselves

RayRedGoose commented:
I found some more bugs in our app and fixed them:
- after login is clicked and app fetches user and data, there is a possibility to start signup at the same time. 
- your issue with questions in round 2 also exists in a final round. I fixed it by adding hyphens and `overflow-wrap: brake-word`.
- duplicated components with the same content but different names. I deleted an unused component.

RayRedGoose commented:
Also, I updated snapshots for failed tests.

RayRedGoose commented:
> Also what do you think about allowing the typed input to remain as it goes from round 1 to 2, 2 to 3 for RoundTwo. Part of me was thinking of clearing the input automatically, but then another part of me was thinking that if they had typed up a question they really liked and then it was removed that might be worse than them having to delete it themselves

I added clear input functionality to useEffect before it's going to the next sentence. Now, we have empty input for every new sentence.

JamesRexMiller4 commented on src/Components/SignUpForm/SignUpForm.test.tsx:
```diff
@@ -3,83 +3,62 @@
+  //
+  //       expect(wrapper.find('button').getDOMNode().disabled).toEqual(false);
+  //   });
+  // });
```
I found some commented out code when I redid the ActionsField component on my last PR. I am cool with merging this in, but lets maybe take a day to go through and refactor our tests, and make sure every component has at least one test for it

JamesRexMiller4 commented on src/Components/BrainstormCard/BrainstormCard.test.tsx:
```diff
@@ -8,10 +8,16 @@ describe('BrainstormCard component', () => {
   const mockProps = {
     id: 1,
```
Is this Prettier making these changes? I don't mind reviewing files, but this seems a little superfluous

JamesRexMiller4 commented on src/Components/GenerateInsights/GenerateInsights.scss:
```diff
@@ -20,6 +20,12 @@
+    -ms-hyphens: auto;
+    hyphens: auto;
+    overflow-wrap: break-word;
+    width: 90%;
```
Amazo! 🚀

JamesRexMiller4 commented on src/Components/LoginForm/LoginForm.test.tsx:
```diff
@@ -4,88 +4,62 @@
   });
 
-
   it('should match the snapshot', () => {
```
Nice refactor, I have been thinking afterEach is more important than beforeEach

JamesRexMiller4 commented:
Nice job reorganizing the files and refactoring here and there!
